### PR TITLE
replace circleci to use github actions

### PR DESCRIPTION
**Pull request by cpanato**
#### Summary
some updates:
- replace circleci to use github actions

the circleci for this repository is already disabled

#### Ticket Link
n/a

### CODE REVIEW

**Comment by cpanato** [comment]
will cherry pick this to the active release branches

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+      working-directory: mattermost-server
+      run: |
+        git checkout $GITHUB_HEAD_REF || git checkout master
+
+    - name: Start containers
+      working-directory: mattermost-server/build
+      env:
+        COMPOSE_PROJECT_NAME: "actions"
+      run: |
+        # cd ../mattermost-server/build
```
Comment (spirosoik) [inline .github/workflows/test.yml:80]: redundant

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+        docker-compose --no-ansi run --rm start_dependencies
```
Comment (spirosoik) [inline .github/workflows/test.yml:81]: ideally all these could be `start_test_containers.sh` with bash and set the right flags for trace the errors. not a blocker though

**Review by spirosoik — COMMENTED** [review]
minor comments

File: .github/workflows/test.yml (hunk 1 above)
Comment (cpanato) [reply .github/workflows/test.yml:80]: this actually is a left over thanks for the catch

File: .github/workflows/test.yml (hunk 2 above)
Comment (cpanato) [reply .github/workflows/test.yml]: done 
thanks for the review!

**Review by spirosoik — APPROVED** [review]

**Review by isacikgoz — APPROVED** [review]
LGTM, nice work @cpanato 🚀

**Comment by agnivade** [comment]
@cpanato - Is this good to be merged? There are some PRs whose CI is blocked due to this. See #435.

**Comment by cpanato** [comment]
> @cpanato - Is this good to be merged? There are some PRs whose CI is blocked due to this. See #435.

yes

**Pull request merged**